volatility3/framework/plugins/linux/lsblk.py

```diff
@@ -0,0 +1,447 @@
+            sd_fops = vmlinux.object_from_symbol("sd_fops")
+
+            if gendisk.fops == sd_fops.vol.offset:
+                scsi_disk = vmlinux.object(
+                    object_type="scsi_disk",
+                    offset=gendisk.private_data,
+                    absolute=True,
+                )
+                return self._get_scsi_device_type(scsi_disk.device.type)
+
+            elif gendisk.fops == sr_bdops.vol.offset:
+                scsi_cd = vmlinux.object(
+                    object_type="scsi_cd",
+                    offset=gendisk.private_data,
+                    absolute=True,
+                )
+                return self._get_scsi_device_type(scsi_cd.device.type)
+
+        return "disk"
+
+    def _get_scsi_device_type(self, type_code):
+        """
+        SCSI device types.  Copied almost as-is from kernel header
+        (include/scsi/scsi_proto.h)
+        """
+        type_map = {
+            0x00: "disk",
+            0x01: "tape",
+            0x02: "printer",
+            0x03: "processor",
+            0x04: "worm",
+            0x05: "rom",
+            0x06: "scanner",
+            0x07: "mo-disk",
+            0x08: "changer",
+            0x09: "comm",
+            0x0C: "raid",
+            0x0D: "enclosure",
+            0x0E: "rbc",
+            0x11: "osd",
+            0x14: "zbc",
+            0x1E: "wlun",
+            0x7F: "no-lun",
+        }
+
+        return type_map.get(type_code, "disk")
+
+    def _get_mountpoint(self, vmlinux, block_device):
+        # check if the device is a swap device
+        S_SWAPFILE = 1 << 8
+        if block_device.bd_inode.i_flags & S_SWAPFILE:
+            return "[SWAP]"
+
+        result = ""
+        # in linux version 6.6, bd_super field is removed from block_device
+        if vmlinux.get_type("block_device").has_member("bd_super"):
+            if block_device.bd_super == 0:
+                return result
+            super_block = block_device.bd_super
+        else:
+            if block_device.bd_holder == 0:
+                return result
+            super_block = vmlinux.object(
+                object_type="super_block",
+                offset=block_device.bd_holder,
+                absolute=True,
+            )
+            if block_device.vol.offset != super_block.s_bdev:
+                return result
+
+        for mount in super_block.s_mounts.to_list(
+            vmlinux.symbol_table_name + constants.BANG + "mount", "mnt_instance"
+        ):
+            while mount.mnt_parent != mount:
+                dentry = mount.mnt_mountpoint
+                mount = mount.mnt_parent
+                while mount.get_mnt_root() != dentry:
+                    result = "/" + dentry.d_name.name_as_str() + result
+                    dentry = dentry.d_parent
+            if result:
+                return result
+            return "/"
+
+        return result
+
+    def run(self):
+
+        columns = [
+            ("NAME", str),
```

Comment:
Why are we shouting all the column names?  5;P  These can just be normal capitalized values like other plugins.

Reply:
It is matching the lsblk command on the live system:

```
$ lsblk
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
sda      8:0    0   40G  0 disk 
├─sda1   8:1    0   39G  0 part /
├─sda2   8:2    0    1K  0 part 
└─sda5   8:5    0  975M  0 part [SWAP]
sdb      8:16   0   40G  0 disk 
└─sdb1   8:17   0   40G  0 part /mnt/samples
sr0     11:0    1 1024M  0 rom  
```